scripts/partition_manager.py

```diff
@@ -220,84 +283,63 @@ def get_header_guard_end(filename):
+    config_lines = get_config_lines(pm_config, "#define ", " ")
 
-    # Store complete size/address configuration to all input paths
-    for area_name, area_props in adr_map.items():
-        if 'out_dir' in area_props.keys():
-            write_pm_config_to_file(path.join(area_props['out_dir'], pm_config_file), area_props['pm_config'])
-
-    # Store to root app, but
-    if not only_sub_image_is_being_built(adr_map, app_output_dir):
-        write_pm_config_to_file(path.join(app_output_dir, pm_config_file), adr_map['app']['pm_config'])
-
-
-def write_pm_config_to_file(pm_config_file_path, pm_config):
-    with open(pm_config_file_path, 'w') as out_file:
-        out_file.write('\n'.join(pm_config))
+    for image_name, dirs in input_config.items():
+        image_config_lines = list.copy(config_lines)
+        image_config_lines.append("#define PM_ADDRESS 0x%x" % pm_config[image_name]['address'])
+        image_config_lines.append("#define PM_SIZE 0x%x" % pm_config[image_name]['size'])
+        image_config_lines.insert(0, get_header_guard_start(pm_config_file))
+        image_config_lines.append(get_header_guard_end(pm_config_file))
+        write_config_lines_to_file(path.join(dirs['out_dir'], pm_config_file), image_config_lines)
 
 
-def write_kconfig_file(adr_map, app_output_dir):
+def write_kconfig_file(pm_config, input_config):
     pm_kconfig_file = "pm.config"
-    config_lines = get_config_lines(adr_map, "", "=")
+    config_lines = get_config_lines(pm_config, "", "=")
 
     # Store out dir and build dir
-    for area_name, area_props in adr_map.items():
-        if 'out_dir' in area_props.keys():
-            config_lines.append('PM_%s_OUT_DIR="%s"' % (area_name.upper(), area_props['out_dir']))
-            config_lines.append('PM_%s_BUILD_DIR="%s"' % (area_name.upper(), area_props['build_dir']))
-
-    # Store output dir to app
-    config_lines.append('PM_APP_OUT_DIR "%s"' % app_output_dir)
+    for image_name, dirs in input_config.items():
+        config_lines.append('PM_%s_OUT_DIR="%s"' % (image_name.upper(), dirs['out_dir']))
+        config_lines.append('PM_%s_BUILD_DIR="%s"' % (image_name.upper(), dirs['build_dir']))
 
     # Store complete size/address configuration to all input paths
-    for area_name, area_props in adr_map.items():
-        if 'out_dir' in area_props.keys():
-            write_pm_config_to_file(path.join(area_props['out_dir'], pm_kconfig_file), config_lines)
+    for dirs in input_config.values():
+        write_config_lines_to_file(path.join(dirs['out_dir'], pm_kconfig_file), config_lines)
+
 
-    # Store to root app
-    if not only_sub_image_is_being_built(adr_map, app_output_dir):
-        write_pm_config_to_file(path.join(app_output_dir, pm_kconfig_file), config_lines)
+def write_yaml_out_file(pm_config, input_config, filename):
+    yamldump = yaml.dump(pm_config)
+    for dirs in input_config.values():
+        write_config_lines_to_file(path.join(dirs['out_dir'], filename), [yamldump])
```

Comment:
Looking at the output, I'm not able to deduce what partitions are phony and not, any comments to this? For this to be useful for other scripts it would be necessary to identify what partitions are phony and not.